src/uPlot.js

```diff
@@ -1746,12 +1801,12 @@ export default function uPlot(opts, data, then) {
 	}
 
 	function mouseDown(e, src, _x, _y, _w, _h, _i) {
-		if (e == null || filtMouse(e)) {
+		if (src != null || filtMouse(e)) {
 			dragging = true;
 
-			cacheMouse(e, src, _x, _y, _w, _h, _i, true, true);
+			cacheMouse(e, src, _x, _y, _w, _h, _i, true, false);
 
-			if (select.show && (drag.x || drag.y))
+			if (select.show && drag.setScale && (drag.x || drag.y))
```

Comment:
why does selection clearing during mousedown now depend on whether there's a setscale call at mouseup? i think these continue to be orthogonal concerns and should not be conflated. is there a reason for this or is these leftovers from before?

Reply:
leftover